packages/opentelemetry-core/src/trace/instrumentation/BasePlugin.ts

```diff
@@ -32,13 +55,77 @@ export abstract class BasePlugin<T> implements Plugin<T> {
     this._moduleExports = moduleExports;
     this._tracer = tracer;
     this._logger = logger;
+    if (config && config.basedir) this.basedir = config.basedir as string;
+    this._internalFilesExports = this._loadInternalFiles();
     return this.patch();
   }
 
   disable(): void {
     this.unpatch();
   }
 
+  private _loadInternalFiles(): ModuleExportsMapping {
+    let extraModules: ModuleExportsMapping = {};
+    if (this._internalFilesList) {
+      this._logger.debug('loadInternalFiles %o', this._internalFilesList);
+      Object.keys(this._internalFilesList).forEach(versionRange => {
+        if (
+          this.version &&
+          this.moduleName &&
+          semver.satisfies(this.version, versionRange)
+        ) {
+          if (Object.keys(extraModules).length > 0) {
+            this._logger.warn(
+              'Plugin for %s@%s, has overlap version range (%s) for internal files: %o',
+              this.moduleName,
+              this.version,
+              versionRange,
+              this._internalFilesList
+            );
+          }
+          extraModules = this._loadInternalModuleFiles(
+            this._internalFilesList![versionRange],
+            this.basedir
+          );
+        }
+      });
+    }
+    if (Object.keys(extraModules)) {
+      this._logger.debug(
+        'No internal files could be loaded for %s@%s',
+        this.moduleName,
+        this.version
+      );
+    }
+    return extraModules;
+  }
+
+  private _loadInternalModuleFiles(
+    extraModulesList: ModuleNameToFilePath,
+    basedir: string
+  ): ModuleExportsMapping {
+    const extraModules: ModuleExportsMapping = {};
+    if (extraModulesList) {
```

Comment:
Could this be re-written as a guard clause to avoid the extra nesting, so something like this:

````
if (!extraModulesList) return {};
const extraModules: ModuleExportsMapping = {};
....
return extraModules;
```

Reply:
Good idea. Refactored to make this change